### PR TITLE
split out jenkins build and push, as we may want to build a branch but not push.  Also made it more easily reusable by templating container repos instead of just tags

### PR DESCRIPTION
**Pull request by allandegnan**
This is mostly just a cleanup so we can support additional / better functionality in the future and reuse this part of the repo for other forms

### CODE REVIEW

**Comment by easternbloc** [comment]
such title!

File: Jenkinsfile [hunk 1]
```diff
@@ -1,37 +1,40 @@
...
-                passwordVariable: 'PASS', usernameVariable: 'USER']]) {
-            sh "docker login -e devops@digital.homeoffice.gov.uk -u ${env.USER} -p ${env.PASS} quay.io"
-            sh "docker push ${app_repo}"
-        }
-       
-        return app_tag
+    return app_tag
+}
+
+def push( app_tag ) {
```
Comment (JoeChapman) [inline Jenkinsfile]: extra spaces around parameter

File: Jenkinsfile [hunk 2]
```diff
@@ -41,11 +44,23 @@ def deploy(environment, app_tag) {
 def runIntegrationTests() {
```
Comment (JoeChapman) [inline Jenkinsfile]: noop?

File: Jenkinsfile [hunk 3]
```diff
@@ -41,11 +44,23 @@ def deploy(environment, app_tag) {
...
 }
 
-stage "Build (includes unit tests)"
-    app_tag=build()
+# main logic
+
+repository = 'quay.io'
+namespace = 'ukhomeofficedigital'
+container_name = 'gro-form'
```
Comment (JoeChapman) [inline Jenkinsfile]: can this be parameterized?

File: Jenkinsfile [hunk 4]
```diff
@@ -41,11 +44,23 @@ def deploy(environment, app_tag) {
...
+repo_name = repository + '/' + namespace + '/' container_name
+
+node() {
+    stage "Build (includes unit tests)"
+        app_tag=build( repo_name )
```
Comment (JoeChapman) [inline Jenkinsfile]: spaces?

File: Jenkinsfile [hunk 5]
```diff
@@ -41,11 +44,23 @@ def deploy(environment, app_tag) {
...
+
+    stage "Push"
+        push( app_tag )
```
Comment (JoeChapman) [inline Jenkinsfile]: spaces?

File: deploy/services/k8resources/redis/redis-rc.yaml [hunk 6]
```diff
@@ -13,7 +13,7 @@ spec:
     spec:
       containers:
       - name: redis
-        image: quay.io/ukhomeofficedigital/redis:${ redis_tag }
+        image: ${ redis_tag }
```
Comment (JoeChapman) [inline deploy/services/k8resources/redis/redis-rc.yaml]: DO you like spaces or not?

**Comment by JoeChapman** [comment]
Can you change the dir name from kube to deploy please?

**Comment by JoeChapman** [comment]
@allandegnan what's happening with this?

File: Jenkinsfile (hunk 3 above)
Comment (allandegnan) [inline Jenkinsfile]: Well it is parameterized but you mean, you want it pulled from another source?

**Comment by JoeChapman** [comment]
!conflicts!

**Comment by JoeChapman** [comment]
@allandegnan this needs rebasing

**Comment by JoeChapman** [comment]
@allandegnan @jon-shanks @easternbloc should this be closed? what's happening with this pr?

**Comment by easternbloc** [comment]
one would hope a large rebase to get some decent commit messages in there...

**Comment by allandegnan** [comment]
@jon-shanks wants us to seperate the deploy code from the app; so likely this should be going elsewhere.

**Comment by JoeChapman** [comment]
@allandegnan can I close this?

**Comment by JoeChapman** [comment]
@allandegnan Close?

**Comment by JoeChapman** [comment]
@allandegnan close?

**Pull request closed without merging**